src/distributions/float.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// https://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Basic floating-point number distributions
+
+
+/// A distribution to sample floating point numbers uniformly in the open
+/// interval `(0, 1)` (not including either endpoint).
+///
+/// See also: [`Closed01`] for the closed `[0, 1]`; [`Uniform`] for the
+/// half-open `[0, 1)`.
+///
+/// # Example
+/// ```rust
+/// use rand::{weak_rng, Rng};
+/// use rand::distributions::Open01;
+///
+/// let val: f32 = weak_rng().sample(Open01);
+/// println!("f32 from (0,1): {}", val);
+/// ```
+///
+/// [`Uniform`]: struct.Uniform.html
+/// [`Closed01`]: struct.Closed01.html
+#[derive(Clone, Copy, Debug)]
+pub struct Open01;
+
+/// A distribution to sample floating point numbers uniformly in the closed
+/// interval `[0, 1]` (including both endpoints).
+///
+/// See also: [`Open01`] for the open `(0, 1)`; [`Uniform`] for the half-open
+/// `[0, 1)`.
+///
+/// # Example
+/// ```rust
+/// use rand::{weak_rng, Rng};
+/// use rand::distributions::Closed01;
+///
+/// let val: f32 = weak_rng().sample(Closed01);
+/// println!("f32 from [0,1]: {}", val);
+/// ```
+///
+/// [`Uniform`]: struct.Uniform.html
+/// [`Open01`]: struct.Open01.html
+#[derive(Clone, Copy, Debug)]
+pub struct Closed01;
+
+
+macro_rules! float_impls {
+    ($mod_name:ident, $ty:ty, $mantissa_bits:expr, $method_name:ident) => {
+        mod $mod_name {
```

Comment:
Why create modules here?

Reply:
I simply moved this code from `rand_impls.rs`. But I think it does have a purpose — `SCALE` is a local constant.

Reply:
Good reason. Then I feel free to remove them :smile: